model/testing/src/icon4py/model/testing/helpers.py

```diff
@@ -77,6 +107,9 @@ class Output:
 
 
 def _test_validation(self, grid, backend, input_data):
```

Comment:
As far as I can see `backend` is not a `str` here, but a `Backend | None`. Which means `match_marker` accepts `backend: Backend | None`. Which would allow to delete the line in the comment (1).